src/fcntl.rs

```diff
@@ -489,6 +489,8 @@ pub enum FcntlArg<'a> {
     F_GETPIPE_SZ,
     #[cfg(any(target_os = "linux", target_os = "android"))]
     F_SETPIPE_SZ(c_int),
+    #[cfg(any(target_os = "netbsd", target_os = "macos", target_os = "ios"))]
+    F_GETPATH(Vec<u8>),
```

Comment:
For arguments like `F_GETXXX`, the associated type should be a mutable reference:

```rust
F_GETPATH(&'a mut Vec<u8>),
```

-----

BTW, do you think it would be better to use a `PathBuf` here since this argument semantically is a path:

```rust
F_GETPATH(&'a mut PathBuf),
```

This approach has an obvious drawback that we can not alter the buffer in place, so we probably have to do another allocation:

```rust
F_GETPATH(path /*: &'a mut PathBuf*/) => {
    let mut another__buf: Vec<u8> =  Vec::with_capacity(libc::PATH_MAX);
    another_buf.resize(libc::PATH_MAX, 0);
    let res = libc::fcntl(fd, libc::F_GETPATH, another_buf.as_ptr().cast());
   
    if res == -1 {
        res
    } else {
        let actual_len = another_buf.iter().position(|byte| *byte == 0).unwrap();
        another_buf.resize(actual_size, 0) // 0 doesn't matter here as `actual_len` is smaller than `PATH_MAX`
        let os_string = OsString::from_vec(another_buf);
        let path_buf = Path::from(os_string);
        *path = path_buf;
        
         res
    }
},
```

Reply:
It we were to choose the PathBuf route, I think

- `let mut another__buf = vec![0; libc::PATH_MAX as usize]` is simpler (and faster) ?
- `libc::fcntl(fd, libc::F_GETPATH, another_buf.as_ptr().cast())`, not sure cast() is necessary here.

Reply:
> let mut another__buf = vec![0; libc::PATH_MAX as usize] is simpler (and faster) ?

Yep, this is better

> libc::fcntl(fd, libc::F_GETPATH, another_buf.as_ptr().cast()), not sure cast() is necessary here.

indeed, it should be removed